### PR TITLE
Fast unstake

### PR DESCRIPTION
**Pull request by uk4z**

### CODE REVIEW

**Review by uk4z — COMMENTED** [review]
lgtm

File: canister_ids.json [hunk 1]
```diff
@@ -25,5 +25,8 @@
 	},
 	"staging_frontend": {
 		"ic": "47pxu-byaaa-aaaap-ahpsa-cai"
+	},
+	"icpswap": {
```
Comment (leo-ship-it) [inline canister_ids.json:29]: Give more context, like nicp-icp-pool-icpswap

File: src/lib/authentification.ts [hunk 2]
```diff
@@ -162,6 +168,7 @@ export async function internetIdentityLogout() {
 	const autClient = await AuthClient.create();
 	await autClient.logout();
 }
+import { AccountIdentifier, type Account, SubAccount } from '@dfinity/ledger-icp';
```
Comment (leo-ship-it) [inline src/lib/authentification.ts:171]: Let's move this import to the top for consistency.

File: src/lib/state.ts [hunk 3]
```diff
@@ -64,39 +61,11 @@ const DAO_SHARE = BigNumber(0.1);
...
-	);
-
-	return Actor.createActor(idl, {
-		canisterId,
-		agent: await dummyAgent
-	}) as any;
-}
-
-export async function fetchWtnAllocation(principal: Principal): Promise<bigint | undefined> {
+export async function fetchWtnAllocation(
```
Comment (leo-ship-it) [inline src/lib/state.ts:64]: Tres bien!

File: src/routes/stake/Stake.svelte [hunk 4]
```diff
@@ -202,21 +199,23 @@
...
 		position: relative;
+		gap: 1em;
 	}
 
 	.error {
-		color: red;
+		display: flex;
+		align-items: center;
+		color: var(--title-color);
+		gap: 10px;
```
Comment (leo-ship-it) [inline src/routes/stake/Stake.svelte:211]: I think this is too much on mobile, 
```suggestion
		gap: 0.2em;
```

File: src/routes/stake/Unstake.svelte [hunk 5]
```diff
@@ -14,22 +23,33 @@
...
 	let invertExchangeRate = false;
+	let isFastUnstake = true;
 	let exchangeRate: BigNumber;
-	let totalIcpDeposited: BigNumber;
 	let minimumWithdraw: BigNumber;
+	let fastUnstakeAmount = BigNumber(0);
+	let showFailedHelp = false;
+	let showImmediateHelp = false;
+	let showDelayedHelp = false;
+	const FEE = 10_000n;
```
Comment (leo-ship-it) [inline src/routes/stake/Unstake.svelte:52]:
```suggestion
	const DEFAULT_LEDGER_FEE = 10_000n;
```

File: src/routes/stake/Unstake.svelte [hunk 6]
```diff
@@ -77,11 +97,228 @@
...
+				console.log('[ICPSwap deposit] error:', error[errorKey]);
+				toasts.add(Toast.error('Failed to deposit nICP on ICPSwap. Please try again.'));
+				break;
+		}
+		return depositResult;
+	};
+
+	const swapInFastUnstake = async (amountIn: string, amountOut: string) => {
+		const swapResult = await $canisters.icpswap.swap({
+			amountIn: amountIn.toString(),
```
Comment (leo-ship-it) [inline src/routes/stake/Unstake.svelte:141]: Here you don't deduct the transfer fee but you do in `depositInFastUnstake`. Try to make it consistent, either way is fine.

File: src/routes/stake/Unstake.svelte [hunk 7]
```diff
@@ -77,11 +97,228 @@
...
+				break;
+		}
+		return swapResult;
+	};
+
+	const withdrawInFastUnstake = async (amountToWithdrawE8s: bigint) => {
+		const withdrawResult = await $canisters.icpswap.withdraw({
+			fee: FEE,
+			token: CANISTER_ID_ICP_LEDGER,
+			amount: amountToWithdrawE8s
```
Comment (leo-ship-it) [inline src/routes/stake/Unstake.svelte:163]: Here you don't deduct as well.

**Review by leo-ship-it — APPROVED** [review]
Thanks for all the hard work!!!

File: src/routes/stake/Unstake.svelte (hunk 5 above)
Comment (uk4z) [reply src/routes/stake/Unstake.svelte:52]: done

File: src/routes/stake/Stake.svelte (hunk 4 above)
Comment (uk4z) [reply src/routes/stake/Stake.svelte:211]: done

File: src/lib/authentification.ts (hunk 2 above)
Comment (uk4z) [reply src/lib/authentification.ts:171]: unused import my bad. Got rid of it

**Pull request merged**